doc/source/whatsnew/v0.21.0.txt

```diff
@@ -91,6 +91,7 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+- Bug in get_indexer_non_unique inconsistent return type with get_indexer (:issue:`16819`)
```

Comment:
``Index.get_indexer_non_unique()`` now returns a ``ndarray`` indexer rather than an ``Index``; this is consistent with ``Index.get_indexer()``.

put in API breaking changes section.